sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -358,21 +518,168 @@ private case class GpuParquetFileFilterHandler(@transient sqlConf: SQLConf) exte
     }
   }
 
-  @scala.annotation.nowarn(
-    "msg=constructor ParquetFileReader in class ParquetFileReader is deprecated"
-  )
+  private def addNamesAndCount(names: ArrayBuffer[String], children: ArrayBuffer[Int],
+      name: String, num_children: Int): Unit = {
+    names += name
+    children += num_children
+  }
+
+  // TODO this needs to move into spark.sql package of some kind to get the right type
+  // checks, for now it is a bit hacky
+  private def depthFirstNamesHelper(schema: DataType, elementName: String, makeLowerCase: Boolean,
+      names: ArrayBuffer[String], children: ArrayBuffer[Int]): Unit = {
+    val name = if (makeLowerCase) {
+      elementName.toLowerCase(Locale.ROOT)
+    } else {
+      elementName
+    }
+    schema match {
+      case cst: StructType =>
+        addNamesAndCount(names, children, name, cst.length)
+        cst.fields.foreach { field =>
+          depthFirstNamesHelper(field.dataType, field.name, makeLowerCase, names, children)
+        }
+      case _: NumericType | BinaryType | BooleanType | DateType | TimestampType | StringType =>
+        addNamesAndCount(names, children, name, 0)
+      case at: ArrayType =>
+        addNamesAndCount(names, children, name, 1)
+        // TODO are element and list hard coded in parquet? or is it something that we should skip
+        addNamesAndCount(names, children, "list", 1)
+        depthFirstNamesHelper(at.elementType, "element", makeLowerCase, names, children)
+      case mt: MapType =>
+        addNamesAndCount(names, children, name, 1)
+        // TODO are key_value, key, and value hard coded in parquet? or is it something that we
+        //  should skip
+        addNamesAndCount(names, children, "key_value", 2)
+        depthFirstNamesHelper(mt.keyType, "key", makeLowerCase, names, children)
+        depthFirstNamesHelper(mt.valueType, "value", makeLowerCase, names, children)
+      case other =>
+        throw new UnsupportedOperationException(s"Need some help here $other...")
+    }
+  }
+
+  def depthFirstNames(schema: StructType, makeLowerCase: Boolean): (Array[String], Array[Int]) = {
+    withResource(new NvtxRange("prepare schema", NvtxColor.WHITE)) { _ =>
+      // Initialize them with a quick length for non-nested values
+      val names = new ArrayBuffer[String](schema.length)
+      val children = new ArrayBuffer[Int](schema.length)
+      schema.fields.foreach { field =>
+        depthFirstNamesHelper(field.dataType, field.name, makeLowerCase, names, children)
+      }
+      (names.toArray, children.toArray)
+    }
+  }
+
+  def readAndFilterFooter(
+      file: PartitionedFile,
+      conf : Configuration,
+      readDataSchema: StructType,
+      filePath: Path): ParquetFooter = {
+    val (names, children) = depthFirstNames(readDataSchema, !isCaseSensitive)
+    val fs = filePath.getFileSystem(conf)
+    val stat = fs.getFileStatus(filePath)
+    // TODO should we do some checks here???
```

Comment:
Would be good to elaborate on the type of checks desired (if not outright implement them).  Some minimal sanity checking is done below (i.e.: file length).